### PR TITLE
2748 Data Table support for any view in project including CompositeViews. Data Tables used in new Studio dashboards

### PR DESCRIPTION
**Pull request by nicwells**
Fixes https://github.com/BlueBrain/nexus/issues/2748

## Description

Data Table component has been modified to support specifying any view in a project. This includes Composite Views with a specified projection or all projections. Studios have been modified to use the Data Table component for any newly created dashboards. Existing dashboards will continue to use existing components. A message box in the Edit Dashboard modal for a dashboard using the existing component indicates that to upgrade to the new structure, delete and re-create the dashboard (as illustrated in the screenshot below).

Screen Shot 2021-09-27 at 17 23 38-fullpage

If there are any existing dashboards in a studio then the existing option to "Select View for the Dashboards(s)" which allows selection of the default Sparql or ES view will be displayed, otherwise it won't (as illustrated below).

Screen Shot 2021-09-27 at 17 50 58-fullpage

The new dashboard using the Data Table component is illustrated below.
Screen Shot 2021-09-27 at 18 07 49-fullpage

The Data Table component View select now supports selecting any view defined in the project. If the selected view is a Composite View then the Projection option will be displayed allowing the user to select which projection to target. There is also an 'All' option to search across all projections on a view.

Screen Shot 2021-09-27 at 18 09 26-fullpage

The Data Table component has been modified to take additional parameters to alter it's behaviour and the available options to make it suitable for use in both Workflows (where it was used previously) and Studios.

## How has this been tested?

Tested in local environment pointing to staging environment. Tested displaying and editing existing dashboards as well as new dashboards.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added necessary unit and integration tests.
- [x] I have added screenshots (if applicable), in the comment section.

### CODE REVIEW

**Review by dhaneshnm — DISMISSED** [review]

File: src/shared/hooks/useAccessDataForTable.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export type TableResource = Resource<{
     '@id': string;
   };
   view: string;
+  projection?: { '@id'?: string; '@type': string };
```
Comment (Stafeeva) [inline src/shared/hooks/useAccessDataForTable.ts:24]: is it possible to use the Projection type here that you already have?

File: src/shared/hooks/useAccessDataForTable.ts [hunk 2]
```diff
@@ -130,7 +135,7 @@ export function parseESResults(result: any) {
 
     return {
       ...resource,
-      key: hit._source._self,
+      // key: hit._source._self,
```
Comment (Stafeeva) [inline src/shared/hooks/useAccessDataForTable.ts:138]: forgotten comment?

File: src/subapps/studioLegacy/containers/DashboardListContainer.tsx [hunk 3]
```diff
@@ -107,6 +117,98 @@ const DashboardList: React.FunctionComponent<DashboardListProps> = ({
...
+          projectLabel,
+          encodeURIComponent(workspaceId),
+          workspace._rev,
+          {
+            ...workspaceSource,
+            dashboards: [
+              ...workspaceSource.dashboards,
+              {
+                dashboard: dashboard['@id'],
+                // view: 'graph',
```
Comment (Stafeeva) [inline src/subapps/studioLegacy/containers/DashboardListContainer.tsx:174]: do we want this comment? :)

Comment (Stafeeva) [inline src/subapps/studioLegacy/containers/DashboardListContainer.tsx:174]: do we want this comment? :)

File: src/shared/containers/DataTableContainer.tsx [hunk 4]
```diff
@@ -235,24 +296,26 @@ const DataTableContainer: React.FC<DataTableProps> = ({
...
+          <div>
+            <Button
+              block
+              danger
+              icon={<DeleteOutlined />}
+              onClick={confirmDeprecate}
+            >
+              Delete
+            </Button>
+            <Modal></Modal>
```
[the remark this answers is not in the document]
Comment (nicwells) [reply src/shared/containers/DataTableContainer.tsx:309]: Just to check you're paying attention :)

**Review by Stafeeva — APPROVED** [review]

**Pull request closed without merging**